libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2078,6 +2083,76 @@ void TypeChecker::typeCheckABIEncodeFunctions(
 	}
 }
 
+#include <typeindex>
+#include <typeinfo>
+void TypeChecker::typeCheckABIEncodeCheckedfunction(FunctionCall const& _functionCall)
```

Comment:
Maybe the function name is better readable if renamed to `typeCheckABIEncodeCheckedFunction`